### PR TITLE
refactor: search-links

### LINKED ISSUE #104: When doing a search for a component, only open in a new tab if you're on the `/editor` or `/ai` page (where you might have changes you want to save)

### PR DESCRIPTION
**Pull request by RohittCodes**
fixes #104

### CODE REVIEW

File: src/components/SearchComponent.tsx [hunk 1]
```diff
@@ -52,6 +54,10 @@ const SearchComponent: React.FC<SearchComponentProps> = ({
     }
   }, [])
 
+  const shouldOpenInNewTab = () => {
```
Comment (seveibar) [inline src/components/SearchComponent.tsx:57]: This should be a variable not a function

**Review by seveibar — CHANGES_REQUESTED** [review]
Is there a way to do this that uses the Link element

**Comment by seveibar** [comment]
A lot of the code is repeated in the ternary, why not just have the prop to Link change? Does it not work?

**Comment by RohittCodes** [comment]
Link component from the wouter library doesn't handle `targets` natively, it loads the content in the same tab as part of SPA navigation. I’ve tried using anchor tags too, but I noticed that they trigger a full-page reload unless handled by the router so came up with this @seveibar.

**Comment by seveibar** [comment]
@RohittCodes ok, if you fix the var/func issue then ill approve

File: src/components/SearchComponent.tsx [hunk 2]
```diff
@@ -52,6 +54,8 @@ const SearchComponent: React.FC<SearchComponentProps> = ({
     }
   }, [])
 
+  const openInNewTab = location === "/editor" || location === "/ai"
```
Comment (seveibar) [inline src/components/SearchComponent.tsx:57]: don't use verbs for variable names, verbs are for function names. `shouldOpenInNewTab` is a var name

```suggestion
  const shouldOpenInNewTab = location === "/editor" || location === "/ai"
```

**Review by seveibar — CHANGES_REQUESTED** [review]

**Review by seveibar — CHANGES_REQUESTED** [review]
can you add a test for this, sorry it's taking so long but i'm skeptical of the location only having the classname and not query parameters etc.

**Comment by RohittCodes** [comment]
@seveibar should i work on the existing search specs for the tests or should i create a new file?

**Comment by seveibar** [comment]
I generally prefer new files!

File: playwright-tests/search-links.spec.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  const popupPromise = page.waitForEvent("popup")
+  await page.getByRole("link", { name: "testuser/my-test-board A" }).click()
+
+  const popup = await popupPromise
+
+  await popup.waitForLoadState("networkidle")
+
+  const popupUrl = popup.url()
+  console.log(popupUrl)
```
Comment (seveibar) [inline playwright-tests/search-links.spec.ts:26]:
```suggestion
```

**Review by seveibar — APPROVED** [review]

**Pull request merged**